src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -90,6 +95,10 @@ func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 		}
 	}
 
+	if err := r.updateStsImage(ctx, &sts); err != nil {
```

Comment:
You don't need `&sts` as receiver holds LastObservedState.

Reply:
I use `sts` because otherwise I'd just add `sts := r.LastObservedState` in each function, so I preferred to make it obvious that we operate on the sts by passing it around.